### ISSUE TITLE
IDE0079 false positive for IDE0032  using *in-class* `SuppressMessageAttribute` 

### ISSUE THREAD

**Issue by Uzivatel919** [dotnet/roslyn#55529]
**Version Used**: 

Microsoft Visual Studio Community 2019
Version 16.10.4
VisualStudio.16.Release/16.10.4+31515.178

**Sample**:

```c#
public class Test1
{
  // Warning IDE0032 Use auto property
  private readonly int i;
  public int I => i;
}

public class Test2
{
  // Message IDE0079 Remove unnecessary suppression
  [SuppressMessage("Style", "IDE0032:Use auto property", Justification = "<Pending>")]
  private readonly int i;
  public int I => i;
}

public class Test3
{
  // Ok
  #pragma warning disable IDE0079 // Remove unnecessary suppression
  [SuppressMessage("Style", "IDE0032:Use auto property", Justification = "<Pending>")]
  #pragma warning restore IDE0079 // Remove unnecessary suppression
  private readonly int i;
  public int I => i;
}
```

**Expected Behavior**:
No false positive for IDE0032 suppression.

**Actual Behavior**:
False positive presented.

**Sample Project**:
[Ide0079FalsePositive.zip](https://github.com/dotnet/roslyn/files/6962991/Ide0079FalsePositive.zip)

**Comment by akhera99** [comment]
No warning displays in latest version of VS preview 2022 or VS 2019. Closed.

**Comment by Youssef1313** [comment]
@akhera99 I can reproduce in latest preview. Please note that in the project attached, IDE0079 is kept with default severity (Suggestion).

![image](https://user-images.githubusercontent.com/31348972/130499555-0792b3f7-587d-45bb-94e2-fdec38787d0a.png)

I think this should be reopened.

**Comment by akhera99** [comment]
> @akhera99 I can reproduce in latest preview. Please note that in the project attached, IDE0079 is kept with default severity (Suggestion).
> 
> ![image](https://user-images.githubusercontent.com/31348972/130499555-0792b3f7-587d-45bb-94e2-fdec38787d0a.png)
> 
> I think this should be reopened.

I misunderstood the original post. Is the issue that the IDE0079 warning displays? I thought it was correct because the auto property analyzer does not indicate a warning in those cases, so suppressing the auto property analyzer is unnecessary?

**Comment by Youssef1313** [comment]
@akhera99 If that's the case, shouldn't the second suppression with `#pragma` report an unnecessary suppression then? The point is both should have the same behavior. Anything I'm missing?

**Comment by Youssef1313** [comment]
> I thought it was correct because the auto property analyzer does not indicate a warning in those cases

@akhera99 I just tested now to change the severity to warning and the unnecessary suppression is still there:
![fp](https://user-images.githubusercontent.com/31348972/130503270-ffaf6f61-0153-43fb-80d5-92a1868beddc.gif)

**Comment by doesicool** [comment]
I know VB.NET is more or less dead language, but this issue is still not solved for it. When IDE0032 is supressed per attribute i get a IDE0079 false positive (TestClass2). See test code below:
```vb
Imports System.Diagnostics.CodeAnalysis

Namespace Test.IDE0079.VB
	Public Class TestClass1
		' Correct warning IDE0032: Use auto property
		Private ReadOnly _Prop1 As Integer
		Public ReadOnly Property Prop1 As Integer
			Get
				Return _Prop1
			End Get
		End Property
	End Class

	Public Class TestClass2
		' Suppress IDE0032 per attribute
		' IDE0079 false positiv
		<SuppressMessage("Style", "IDE0032:Use auto property")>
		Private ReadOnly _Prop1 As Integer
		Public ReadOnly Property Prop1 As Integer
			Get
				Return _Prop1
			End Get
		End Property
	End Class

	Public Class TestClass3
		' Suppress IDE0032 per preprocessor directive
		' No IDE0079 false positiv
#Disable Warning IDE0032 ' Use auto property
		Private ReadOnly _Prop1 As Integer
#Enable Warning IDE0032 ' Use auto property
		Public ReadOnly Property Prop1 As Integer
			Get
				Return _Prop1
			End Get
		End Property
	End Class
End Namespace
```
(Additional Information: Tested with Visual Studio 2026 Version 18.7.2 in an .NET 8 class library with &lt;AnalysisLevel&gt; = preview-all.)

So please either mark issue with C# label or better reopen as unsolved for VB.NET.

**Comment by CyrusNajmabadi** [comment]
@doesicool Please open a new issue.
